src/node/http_node_client.h

```diff
@@ -57,16 +106,45 @@ namespace ccf
       auto frontend = frontend_opt.value();
       frontend->process(ctx);
 
-      auto rs = ctx->get_response_status();
-
-      if (rs != HTTP_STATUS_OK)
-      {
-        auto ser_res = ctx->serialise_response();
-        std::string str((char*)ser_res.data(), ser_res.size());
-        LOG_FAIL_FMT("Request failed: {}", str);
-      }
-
-      return rs == HTTP_STATUS_OK;
+      threading::retry_until(
```

Comment:
This feels like this shouldn't be in the underlying HTTP client but instead as a wrapper around it.